js/load-image-scale.js

```diff
@@ -96,25 +96,29 @@
     destHeight,
     options
   ) {
-    var ctx = canvas.getContext('2d')
-    if (options.imageSmoothingEnabled === false) {
-      ctx.msImageSmoothingEnabled = false
-      ctx.imageSmoothingEnabled = false
-    } else if (options.imageSmoothingQuality) {
-      ctx.imageSmoothingQuality = options.imageSmoothingQuality
+    try {
+      var ctx = canvas.getContext('2d')
+      if (options.imageSmoothingEnabled === false) {
+        ctx.msImageSmoothingEnabled = false
+        ctx.imageSmoothingEnabled = false
+      } else if (options.imageSmoothingQuality) {
+        ctx.imageSmoothingQuality = options.imageSmoothingQuality
+      }
+      ctx.drawImage(
```

Comment:
@blueimp `drwaImage` may throw several exceptions, like fro too huge images.

In case that happens the Promise should reject or the callback api should have an `type === 'error'`.
How is the best way to move this error up?

Reply:
For the Promise style API, if the result is an error, it should always reject, so your proposed solution is correct.